code/modules/mob/living/brain/brain_item.dm

```diff
@@ -113,24 +113,31 @@
 	if(istype(O, /obj/item/borg/apparatus/organ_storage))
 		return //Borg organ bags shouldn't be killing brains
 
-	if((organ_flags & ORGAN_FAILING) && O.is_drainable() && O.reagents.has_reagent(/datum/reagent/medicine/mannitol)) //attempt to heal the brain
+	var/healby = 2 //heals 2 damage per unit of mannitol
+	var/pour = "carefully pour"
+	var/onto = "onto"
+	if(istype(O, /obj/item/reagent_containers/syringe))
+		healby = 4 //heal 4 damage because PRECISCION
+		pour = "precisely inject"
+		onto = "into"
+	else if(istype(O, /obj/item/reagent_containers/dropper))
```

Comment:
Not sure if I really like this sort of type checking, but unsure what would be a suitable replacement.

Reply:
there is none (currently)